butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/xml/XPathRetrieve.java

```diff
@@ -0,0 +1,178 @@
+package com.paypal.butterfly.utilities.operations.xml;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.xml.namespace.QName;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import com.paypal.butterfly.extensions.api.SingleCondition;
+import com.paypal.butterfly.extensions.api.TUExecutionResult;
+import com.paypal.butterfly.extensions.api.TransformationContext;
+import com.paypal.butterfly.extensions.api.exception.TransformationDefinitionException;
+import com.paypal.butterfly.extensions.api.exception.TransformationUtilityException;
+
+/**
+ * Retrieves data from an XML file by using XPath expressions.
+ * If no element, nor attribute, is found,
+ * {@link com.paypal.butterfly.extensions.api.TUExecutionResult.Type#NULL} is returned.
+ * If the xpath expression won't compile, an error is returned.
+ * <br>
+ * If the file is not a well formed XML file, an error is returned.
+ *
+ * @author spetratos
+ */
+public class XPathRetrieve extends SingleCondition<XPathRetrieve> {
+    private String xpathExpressionString;
+    private XPathExpression xpathExpression;
+    private QName returnDataType;
+
+    private static final String DESCRIPTION = "Retriveve the XML XPath query %s if it exists in XML file %s";
```

Comment:
Please, fix the typo in `Retriveve`. Also, the TU retrieves XML data, not the query itself, correct? Maybe should this be rephrased to something like: "Retrieve XML data based on the given XPath query %s and XML file %s"?